motor/providers/k8s/manifest_provider.go

```diff
@@ -197,29 +198,50 @@ func loadManifestFile(manifestFile string) ([]byte, error) {
 		}
 
 		if fi.IsDir() {
-			// NOTE: we are not using filepath.WalkDir since we do not net recursive walking
-			files, err := ioutil.ReadDir(manifestFile)
-			if err != nil {
-				return nil, err
-			}
-			for i := range files {
-				f := files[i]
-				if f.IsDir() {
-					continue
+			filepath.WalkDir(manifestFile, func(path string, d fs.DirEntry, err error) error {
+				if err != nil {
+					return err
 				}
-				filename := path.Join(manifestFile, f.Name())
 
 				// only load yaml files for now
-				ext := filepath.Ext(filename)
-				if ext == ".yaml" || ext == ".yml" {
-					log.Debug().Str("file", filename).Msg("add file to manifest loading")
-					filenames = append(filenames, filename)
-				} else {
-					log.Debug().Str("file", filename).Msg("ignore file")
+				if !d.IsDir() {
```

Comment:
I think we should add tests for that to make sure it works as expected.

Reply:
I agree. I think a test is needed. Perhaps create a test folder with some manifest, some non-manifest yaml and a readme (or some other random file). Just to make sure we test this bit of code since it is kind of complicated